test/unit-tests/file-system.ts

```diff
@@ -18,6 +18,57 @@ function isOsCaseSensitive(testInjector: IInjector): boolean {
 };
 temp.track();
 
+function createWriteJsonTestCases(): { exists: boolean, text: string, testCondition: string, expectedIndentation: string }[] {
+	return [
+		{
+			exists: true,
+			text: `{\n\t"a" : 5 }`,
+			testCondition: "when the indentation is tab",
+			expectedIndentation: "\t"
+		}, {
+			exists: true,
+			text: `{\n "a" : 5 }`,
+			testCondition: "when the indentation is space",
+			expectedIndentation: " "
+		}, {
+			exists: true,
+			text: `{\n  "a" : 5 }`,
+			testCondition: "when the indentation is two spaces",
+			expectedIndentation: "  "
+		}, {
+			exists: false,
+			text: `{\n "a" : 5 }`,
```

Comment:
Why'd you have any `text` when the file doesn't exist 👀 
I'd prefer this property have an empty value for additional clarity